examples/kitchen-sink.yaml

```diff
@@ -426,3 +426,37 @@ resource:
         - process.command_args
   # Configure the resource schema URL.
   schema_url: https://opentelemetry.io/schemas/1.16.0
+# Configure instrumentation
+instrumentation:
+  # Configure options common to multiple languages
+  common:
+    peer-service-mapping:
+      - peer: foo
+        service: bar
+      - peer: bar
+        service: bat
+    http-capture-headers:
+      client:
+        request:
+          - X-Foo
+        response:
+          - X-Foo-*
```

Comment:
We have guidance [here](https://github.com/open-telemetry/opentelemetry-configuration?tab=readme-ov-file#schema-modeling-guidelines) for naming and semantics of properties which are expected to require pattern matching capabilities.

Reply:
Looking a bit more closely at [Java's documentation for capturing headers](https://opentelemetry.io/docs/languages/java/automatic/configuration/#capturing-http-request-and-response-headers), it doesn't look like wildcards are a feature, so I've just removed that for now.